infrastructure/lib/constructs/compliant-resources.ts

```diff
@@ -64,3 +65,38 @@ export class SecureTable extends cdk.Construct {
     this.table = table;
   }
 }
+
+export interface SecureAPIGatewayProps {
+  /**
+   * The properties to configure the API Gateway
+   */
+  apiGatewayProps: apigw.SpecRestApiProps;
+}
+
+/**
+ * Creates a secure API Gateway with properties enabled for compliance
+ *  - cache enabled and encrypted by default
+ *  - execution logs enabled by default
+ */
+export class SecureAPIGateway extends cdk.Construct {
+  readonly apiGateway: apigw.SpecRestApi;
+  constructor(scope: cdk.Construct, id: string, props: SecureAPIGatewayProps) {
+    super(scope, id);
+
+    const apiGateway = new apigw.SpecRestApi(this, id, {
```

Comment:
@kylelaker-ccpo I agree with making this more flexible depending on the type of api used. The code snippet you gave almost worked, but gave an error. 

![Error 1](https://user-images.githubusercontent.com/89867104/135338122-1f067dfb-af8f-4245-afab-b2d489fed2d5.png)
I tried creating a union type 
``` 
type RestApiTypes =
  | (new (scope: cdk.Construct, id: string, props: apigw.RestApiBaseProps) => apigw.RestApiBase)
  | (new (scope: cdk.Construct, id: string, props: apigw.SpecRestApiProps) => apigw.SpecRestApi);
 ```
that uses the `RestApiBase` or `SpecRestApi` and it worked, but then there is a different error for the `apiDefinition`. I am assuming this is similar to a problem we ran into before and is because the `SecureRestApiProps` has the `apiDefinition` as optional, but the `SpecRestApiProps` has it as being required. 

![Error 2](https://user-images.githubusercontent.com/89867104/135338199-c0f5663a-af84-4f05-9b51-c17ef306ab2c.png)

Am I missing something or is there an approach that I am overlooking to fix this?

If you want, I can go with the fallback option for the moment and we can work through this once you get back to the office. What do you think?

Reply:
Let's go with the latter for now

Reply:
Great write up by the way! Thank you! Definitely makes the problem clear and yeah, let's sync up when we've both got a chance.